indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -346,4 +347,24 @@ public LookupLoadingSpec getLookupLoadingSpec()
   {
     return LookupLoadingSpec.NONE;
   }
+
+  @Override
+  public boolean isReady(TaskActionClient taskActionClient) throws Exception
+  {
+    final boolean isUsingConcurrentLocks = Boolean.TRUE.equals(
+        getContextValue(
+            Tasks.USE_CONCURRENT_LOCKS,
+            Tasks.DEFAULT_USE_CONCURRENT_LOCKS
+        )
+    );
+
+    TaskLockType defaultLockType = !markAsUnused && isUsingConcurrentLocks ? TaskLockType.REPLACE : TaskLockType.EXCLUSIVE;
+
+    TaskLockType actualLockType = getContextValue(Tasks.TASK_LOCK_TYPE, defaultLockType);
+
+    if (markAsUnused && actualLockType != TaskLockType.EXCLUSIVE) {
```

Comment:
`markAsUnused` is undocumented and has been deprecated for a few releases since Druid 28. Instead of expanding on the legacy flag, we can remove it for Druid 31. It should simplify this code and related testing a bit. cc: @kfaraz 

@IgorBerman, would you be interested in doing that in a separate PR?

Reply:
You mean separate PR for removing this flag completely?
I can try that when 31 will be around the corner

Reply:
Yeah, we can remove the deprecated `markAsUnused` parameter from the kill task. IMO, it'd be best if we handle that separately first since this feature expands on the deprecated parameter when it shouldn't be used and needed. 

Alternatively, as part of this patch, we can add a validation check in the kill task constructor, similar to [this](https://github.com/apache/druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java#L144). The `isReady()` method here can then assume that `markAsUnused` is unset or false, so we'd have fewer branches.

The branch for Druid 30 is already cut here: https://github.com/apache/druid/tree/30.0.0. So any changes merged to master from now on will automatically be targeted for Druid 31. Let us know what you think.

Reply:
ok, sure. So I'm removing markAsUnused and any references to it. Hopefully I'll find all of it

Reply:
ok, hopefully removed in all relevant places